### ISSUE TITLE
InventoryClickEvent NPE

### ISSUE THREAD

**Issue by TakoidGit** [JEFF-Media-GbR/Spigot-ChestSort#28]
**Used version**
6.1

**Describe the bug**
An NPE shortly after a new player joined (might be unrelated)

**To Reproduce**
Unknown.

**Expected behavior**
Unsure if a bug occurred or not.

**Additional context**
```
[05:18:36 ERROR]: Could not pass event InventoryClickEvent to ChestSort v6.1
java.lang.NullPointerException: null
        at de.jeffclan.JeffChestSort.JeffChestSortListener.onInventoryClickEvent(JeffChestSortListener.java:311) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor135.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.14.3.jar:git-Paper-131]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.14.3.jar:git-Paper-131]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.14.3.jar:git-Paper-131]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:545) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.PlayerConnection.a(PlayerConnection.java:2341) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.PacketPlayInWindowClick.a(SourceFile:33) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.PacketPlayInWindowClick.a(SourceFile:10) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:18) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.TickTask.run(SourceFile:18) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:127) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:105) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.MinecraftServer.aW(MinecraftServer.java:1002) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.MinecraftServer.executeNext(MinecraftServer.java:995) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:115) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.MinecraftServer.sleepForTick(MinecraftServer.java:979) ~[patched_1.14.3.jar:git-Paper-131]
        at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:912) ~[patched_1.14.3.jar:git-Paper-131]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
```

**Comment by mfnalex** [comment]
Does this always happen?
Actually, line 311 only consists of a closing bracket, I am not sure how a NPE can happen there. Did you maybe /reload your server instead of properly restarting? Please tell me when the bug happens again.

**Comment by mfnalex** [comment]
Please tell me if this happens with the new version 6.2 as well

**Comment by TakoidGit** [comment]
```
[19:29:48] [Server thread/ERROR]: Could not pass event InventoryClickEvent to ChestSort v6.2
java.lang.NullPointerException: null
	at de.jeffclan.JeffChestSort.JeffChestSortListener.onInventoryClickEvent(JeffChestSortListener.java:316) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor135.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.14.3.jar:git-Paper-131]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.14.3.jar:git-Paper-131]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.14.3.jar:git-Paper-131]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:545) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.PlayerConnection.a(PlayerConnection.java:2341) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.PacketPlayInWindowClick.a(SourceFile:33) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.PacketPlayInWindowClick.a(SourceFile:10) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:18) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.TickTask.run(SourceFile:18) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:127) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:105) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.MinecraftServer.aW(MinecraftServer.java:1002) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.MinecraftServer.executeNext(MinecraftServer.java:995) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeAll(IAsyncTaskHandler.java:91) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.MinecraftServer.sleepForTick(MinecraftServer.java:978) ~[patched_1.14.3.jar:git-Paper-131]
	at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:912) ~[patched_1.14.3.jar:git-Paper-131]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
```

**Comment by mfnalex** [comment]
The line `at de.jeffclan.JeffChestSort.JeffChestSortListener.onInventoryClickEvent(JeffChestSortListener.java:316) ~[?:?]` means something went wrong while you tried to Shift-Left-Click an empty slot to sort an inventory.
The line contains the following:
`if(event.getCurrentItem().getType() == Material.AIR) {`
The only thing I can think of is that event.getCurrentItem() somehow is null. However, I have tested this myself with the latest PaperSpigot version and everything works fine.
Are you using the regular and latest PaperSpigot version (build 131)?

**Comment by TakoidGit** [comment]
I believe the issue was because of Spartan anti-cheat.

**Comment by mfnalex** [comment]
That could be possible. However, I can probably create a fix if you still want to use Spartan. Can you please send me a private message at spigotmc.org?

**Comment by mfnalex** [comment]
You might try again with 6.4.1 which *should* fix the issue. However, I couldn't test it since I could not reproduce the error

**Comment by andris155** [comment]
Same problem
I use NoCheatPlus
[```
12:33:32] [Server thread/ERROR]: Could not pass event InventoryClickEvent to ChestSort v6.4
java.lang.NullPointerException: null
	at de.jeffclan.JeffChestSort.JeffChestSortListener.onInventoryClickEvent(JeffChestSortListener.java:316) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor266.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.14.4.jar:git-Paper-135]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.14.4.jar:git-Paper-135]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.14.4.jar:git-Paper-135]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:545) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.PlayerConnection.a(PlayerConnection.java:2310) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.PacketPlayInWindowClick.a(SourceFile:33) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.PacketPlayInWindowClick.a(SourceFile:10) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:23) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.TickTask.run(SourceFile:18) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.MinecraftServer.aX(MinecraftServer.java:1014) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.MinecraftServer.executeNext(MinecraftServer.java:1007) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeAll(IAsyncTaskHandler.java:95) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.MinecraftServer.sleepForTick(MinecraftServer.java:990) ~[patched_1.14.4.jar:git-Paper-135]
	at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:924) ~[patched_1.14.4.jar:git-Paper-135]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

**Comment by mfnalex** [comment]
Fixed in 6.4.1
